src/TextField/TextFieldLabel.jsx

```diff
@@ -95,6 +92,5 @@ const TextFieldLabel = (props) => {
 };
 
 TextFieldLabel.propTypes = propTypes;
-TextFieldLabel.defaultProps = defaultProps;
 
-export default TextFieldLabel;
+export default shouldUpdate(withShallow('style'))(TextFieldLabel);
```

Comment:
I agree 100%. Why don't we close this and I'll revisit if the problem persists once the style memoization improvements are added.